### PR TITLE
adding docs for System.Runtime.CompilerServices

### PR DESCRIPTION
**Pull request by Anipik**
cc @carlossanlop

### CODE REVIEW

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 1]
```diff
@@ -463,9 +463,9 @@
         <Parameter Name="type" Type="System.Type" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="type">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="type">The system-provided types.</param>
+        <summary>This method is used to get an Uninitialized instance of the system-provided types.</summary>
```
Comment (carlossanlop) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]:
```suggestion
        <summary>Returns an uninitialized instance of the system-provided types.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 2]
```diff
@@ -1129,8 +1129,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>This method ensures that there is sufficient stack to execute the average Framework function. If there is not enough stack, then it returns false.</summary>
```
Comment (carlossanlop) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]:
```suggestion
        <summary>Ensures that there is sufficient stack to execute the average Framework function. If there is not enough stack, then it returns false.</summary>
```

Comment (carlossanlop) [reply xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]: What do you mean by Framework? Can a different word be used? It is not very clear.

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 3]
```diff
@@ -463,9 +463,9 @@
         <Parameter Name="type" Type="System.Type" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="type">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="type">The system-provided types.</param>
+        <summary>This method is used to get an Uninitialized instance of the system-provided types.</summary>
+        <returns>System.Object</returns>
```
Comment (carlossanlop) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]: This needs to be a description in plain English like:

```suggestion
        <returns>An object representing the uninitialized system-provided type instance. </returns>
```

**Comment by Jozkee** [comment]
I would suggest to check the [Wiki](https://github.com/dotnet/dotnet-api-docs/wiki) to see boilerplates showing how to document each xml tag.

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 4]
```diff
@@ -1129,8 +1129,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>This method ensures that there is sufficient stack to execute the average Framework function. If there is not enough stack, then it returns false.</summary>
+        <returns><see langword="true" /> if sufficient stack is present; otherwise, <see langword="false" />.</returns>
         <remarks>To be added.</remarks>
```
Comment (Jozkee) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]: I would also add the remark section and the note from `EnsureSufficientExecutionStack` (if applies).
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.ensuresufficientexecutionstack?view=netcore-3.1#remarks

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 5]
```diff
@@ -463,9 +463,9 @@
         <Parameter Name="type" Type="System.Type" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="type">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="type">The system-provided types.</param>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]:
```suggestion
        <param name="type">A system-provided type.</param>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 6]
```diff
@@ -463,9 +463,9 @@
...
+        <summary>Returns an uninitialized instance of the system-provided types.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]:
```suggestion
        <summary>Returns an uninitialized instance of the system-provided type.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml [hunk 7]
```diff
@@ -1129,8 +1129,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Tries to ensure there is sufficient stack to execute the average .net core libraries function.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]:
```suggestion
        <summary>Tries to ensure there is sufficient stack to execute the average .NET Core library function.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml [hunk 8]
```diff
@@ -78,8 +78,8 @@
         <Parameter Name="thrownObject" Type="System.Object" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="thrownObject">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="thrownObject">The exception getting wrapped.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException " /> exception containing the inner exception object.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml]:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException " /> exception that contains the inner exception object.</summary>
```
Better for localization.

**Comment by Anipik** [comment]
@carlossanlop can you merge this one ?

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]: Does this remark apply, @Anipik or did you decide it was not necessary?

File: xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml [hunk 9]
```diff
@@ -78,8 +78,8 @@
         <Parameter Name="thrownObject" Type="System.Object" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="thrownObject">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="thrownObject">The exception getting wrapped.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException " /> exception that contains the inner exception object.</summary>
```
Comment (carlossanlop) [inline xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml]: Double use of "exception" (one is in the type name already) :
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException " /> that contains the inner exception object.</summary>
```

File: xml/System.Runtime.CompilerServices/RuntimeHelpers.xml (hunk 4 above)
Comment (Anipik) [reply xml/System.Runtime.CompilerServices/RuntimeHelpers.xml]: I think its not necessary here. There was nothing extra information that needs to be added in remarks of TryEnsureSufficientExecutionStack which is not already there.

File: xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml [hunk 10]
```diff
@@ -78,8 +78,8 @@
         <Parameter Name="thrownObject" Type="System.Object" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="thrownObject">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="thrownObject">The exception getting wrapped.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException " /> that contains the inner exception object.</summary>
```
Comment (carlossanlop) [inline xml/System.Runtime.CompilerServices/RuntimeWrappedException.xml]: The build had warnings, @Anipik. Looking at the [build Details](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C17%5C500c749d-de78-3d4a-00e4-269d599bf9ec%5CPullRequest%5C201912171732197033-3675%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=rx363BmwfPJGKWodd5rcIkjuWcY8kRDAIPZIDQYjv%2Fw%3D&st=2019-12-17T18%3A04%3A58Z&se=2020-01-17T18%3A09%3A58Z&sp=r), at the very end of the page, it says that one of the crefs was invalid:

```[Warning] 1 invalid cross reference(s) "System.Runtime.CompilerServices.RuntimeWrappedException ".```

It seems there's a trailing space. Can you please fix this so we can get it merged?

```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.RuntimeWrappedException" /> that contains the inner exception object.</summary>
```
